pandas/tests/frame/test_dtypes.py

```diff
@@ -410,41 +409,57 @@ def test_dtypes_gh8722(self):
             result = df.dtypes
             assert_series_equal(result, Series({0: np.dtype('int64')}))
 
-    def test_ftypes(self):
-        frame = self.mixed_float
+    def test_ftypes(self, mixed_float_frame):
+        frame = mixed_float_frame
         expected = Series(dict(A='float32:dense',
                                B='float32:dense',
                                C='float16:dense',
                                D='float64:dense')).sort_values()
         result = frame.ftypes.sort_values()
         assert_series_equal(result, expected)
 
-    def test_astype(self):
-        casted = self.frame.astype(int)
-        expected = DataFrame(self.frame.values.astype(int),
-                             index=self.frame.index,
-                             columns=self.frame.columns)
+    def test_astype_float(self, float_frame):
+        casted = float_frame.astype(int)
+        expected = DataFrame(float_frame.values.astype(int),
+                             index=float_frame.index,
+                             columns=float_frame.columns)
         assert_frame_equal(casted, expected)
 
-        casted = self.frame.astype(np.int32)
-        expected = DataFrame(self.frame.values.astype(np.int32),
-                             index=self.frame.index,
-                             columns=self.frame.columns)
+        casted = float_frame.astype(np.int32)
+        expected = DataFrame(float_frame.values.astype(np.int32),
+                             index=float_frame.index,
+                             columns=float_frame.columns)
         assert_frame_equal(casted, expected)
 
-        self.frame['foo'] = '5'
-        casted = self.frame.astype(int)
-        expected = DataFrame(self.frame.values.astype(int),
-                             index=self.frame.index,
-                             columns=self.frame.columns)
+        float_frame['foo'] = '5'
+        casted = float_frame.astype(int)
+        expected = DataFrame(float_frame.values.astype(int),
+                             index=float_frame.index,
+                             columns=float_frame.columns)
         assert_frame_equal(casted, expected)
 
+    def test_astype_mixed_float(self, mixed_float_frame):
         # mixed casting
         def _check_cast(df, v):
             assert (list({s.dtype.name for
                           _, s in compat.iteritems(df)})[0] == v)
 
-        mn = self.all_mixed._get_numeric_data().copy()
+        casted = mixed_float_frame.reindex(
+            columns=['A', 'B']).astype('float32')
+        _check_cast(casted, 'float32')
+
+        casted = mixed_float_frame.reindex(
+            columns=['A', 'B']).astype('float16')
+        _check_cast(casted, 'float16')
+
+    def test_astype_mixed_type(self, mixed_type_frame):
```

Comment:
i don't want the repeated code.

Reply:
Fine by me, but that means this *should* be done in a separate PR. I'll move the `_check_cast` function to the top of the module.